### PR TITLE
cppcheck: update sha

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/Homebrew/homebrew-core/actions/runs/3310446650/jobs/5464658259
https://github.com/Homebrew/homebrew-core/actions/runs/3410488991/jobs/5673509896
relates to #113717

### CODE REVIEW

File: Formula/cppcheck.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Cppcheck < Formula
   desc "Static analysis of C and C++ code"
   homepage "https://sourceforge.net/projects/cppcheck/"
   url "https://github.com/danmar/cppcheck/archive/2.9.1.tar.gz"
-  sha256 "263fa3da28385f625ceb66857bbf9aff9947ff4ab6feb85c6f17cd7fe4779fe8"
+  sha256 "c166b86aebe80faa08c80196c9156b73e9ffaf8eccd84f9317c51f62577028df"
```
Comment (chenrui333) [inline Formula/cppcheck.rb:5]: @danmar, looks like there is no way to submit an issue, so I will just tag you in this PR to double confirm the re-tag was intentional, thanks!!

Comment (danmar) [reply Formula/cppcheck.rb:5]: oh I did not realize it was used sorry. Yes it was intentional. So I'll be more careful from now on.

Comment (chenrui333) [reply Formula/cppcheck.rb:5]: Thanks for confirming it!

**Comment by chenrui333** [comment]
Actually, we can just do this new bottle in the new 2.9.2 release.

**Pull request closed without merging**